test/unit/validators/RunValidator.spec.ts

```diff
@@ -0,0 +1,240 @@
+    });
+
+    it('should throw an error with language missing', async () => {
+        const req = {
+            body: {
+                source: 'LKJSDFKLMC414CcnBcba12',
+                mode: 'sync',
+                stdin: 'INPUT'
+            }
+        };
+
+        await runValidator.POST(req, res, nextSpy);
+
+        expect(sentStatus).to.equal(400);
+        expect(sentData.err.message).to.equal('"lang" is required');
+        expect(nextSpy.calledOnce).to.be.false;
+    });
+
+    it('should throw an error with incorrect language', async () => {
+        const req = {
+            body: {
+                source: 'LKJSDFKLMC414CcnBcba12',
+                lang: 'wrongLang',
+                mode: 'sync',
+                stdin: 'INPUT'
+            }
+        };
+
+        await runValidator.POST(req, res, nextSpy);
+        // TODO WRONG
+        // expect(sentStatus).to.equal(400);
+        // expect(sentData.err.message).to.equal('"lang" is required');
+        // expect(nextSpy.calledOnce).to.be.false;
+    });
+
+    it('should throw an error when source is missing', async () => {
+        const req = {
+            body: {
+                mode: 'sync',
+                lang: 'cpp',
+                stdin: 'INPUT'
+            }
+        };
+
+        await runValidator.POST(req, res, nextSpy);
+
+        expect(sentStatus).to.equal(400);
+        expect(sentData.err.message).to.equal('"source" is required');
+        expect(nextSpy.calledOnce).to.be.false;
+    });
+
+    it('should throw an error when source is NOT string', async () => {
+        const req = {
+            body: {
+                source: 1212,
+                mode: 'sync',
+                lang: 'cpp',
+                stdin: 'INPUT'
+            }
+        };
+
+        await runValidator.POST(req, res, nextSpy);
+
+        expect(sentStatus).to.equal(400);
+        expect(sentData.err.message).to.equal('"source" must be a string');
+        expect(nextSpy.calledOnce).to.be.false;
+    });
+
+    it('should throw an error when mode is missing', async () => {
+        const req = {
+            body: {
+                source: 'LKJSDFKLMC414CcnBcba12',
+                lang: 'cpp',
+                stdin: 'INPUT'
+            }
+        };
+
+        await runValidator.POST(req, res, nextSpy);
+        // TODO
+        // does not throw error for mode missing
+        expect(nextSpy.calledOnce).to.be.true;
+        // nextSpy is true === next() was called
+    });
+
+    it('should throw an error when mode is NOT string', async () => {
+        const req = {
+            body: {
+                source: 'LKJSDFKLMC414CcnBcba12',
+                lang: 'cpp',
+                mode: 123,
+                stdin: 'INPUT'
+            }
+        };
+
+        await runValidator.POST(req, res, nextSpy);
+        // TODO
+        expect(nextSpy.calledOnce).to.be.false;
+        // error thrown == " mode" must be one of [sync, callback, poll]
```

Comment:
We are relying on Joi errors and as far as this is concerned, we may move to modes being numerical. For e.g. sync = 0, callback = 1